parsl/jobs/strategy.py

```diff
@@ -298,8 +298,8 @@ def _general_strategy(self, executors: List[BlockProviderExecutor], *, strategy_
                     # Scale in for htex
                     if isinstance(executor, HighThroughputExecutor):
                         if active_blocks > min_blocks:
-                            excess_slots = math.ceil(active_slots - (active_tasks * parallelism))
-                            excess_blocks = math.ceil(float(excess_slots) / (tasks_per_node * nodes_per_block))
+                            excess_slots = math.floor(active_slots - (active_tasks * parallelism))
+                            excess_blocks = math.floor(float(excess_slots) / (tasks_per_node * nodes_per_block))
                             excess_blocks = min(excess_blocks, active_blocks - min_blocks)
```

Comment:
I'm sure it's accounted for prior to this if-branch&nbsp;&mdash;&nbsp;and it exists prior to this hunk&nbsp;&mdash;&nbsp;but the first thing that comes to mind is "is it possible for excess_blocks to become negative?)

The if-condition guarantees that `active_blocks - min_blocks` is at least 1, but I'm not clear on the guarantees of `active_slots - (active_tasks * parallelism)`

Reply:
probably many of these values are assumed to be "sensible" and this code will go wrong if not. see for example this recent issue https://github.com/Parsl/parsl/issues/3726